### DISCUSSION TITLE
could not find the following exchanges in community

### DISCUSSION THREAD

**Discussion by oMersa344** [micom-dev/micom#63] [Q&A] [answered]
I’m trying to model community growth (2-3 defined bacterial strains in each community) in rich media that I created to stimulate realistic growth conditions.

the medium I want to model with - 
```
rich_media : {'EX_ca2_m': 1000.0, 'EX_cl_m': 1000.0, 'EX_zn2_m': 1000.0, 'EX_cu2_m': 1000.0, 'EX_fe2_m': 1000.0, 'EX_fe3_m': 1000.0, 'EX_glc__D_m': 10.0, 'EX_h2o_m': 1000.0, 'EX_h_m': 1000.0, 'EX_k_m': 1000.0, 'EX_mg2_m': 1000.0, 'EX_mn2_m': 1000.0, 'EX_mood_m': 1000.0, 'EX_na1_m': 1000.0, 'EX_nh4_m': 1000.0, 'EX_ni2_m': 1000.0, 'EX_o2_m': 1000.0, 'EX_pi_m': 1000.0, 'EX_so4_m': 1000.0, 'EX_malt_m': 10.0, 'EX_hdcea_m': 1000.0, 'EX_ocdcya_m': 1000.0, 'EX_ocdcea_m': 1000.0, 'EX_malttr_m': 10.0, 'EX_man_m': 10.0, 'EX_isomal_m': 10.0, 'EX_maltttr_m': 10.0, 'EX_maltpt_m': 10.0, 'EX_amylopect900_m': 1000.0, 'EX_amylose300_m': 1000.0, 'EX_amylose_m': 1000.0, 'EX_cell500_m': 1000.0, 'EX_cell6_m': 1000.0, 'EX_cell4_m': 1000.0, 'EX_lnlncg_m': 1000.0, 'EX_lnlnca_m': 1000.0, 'EX_starch2_m': 1000.0, 'EX_starch1_m': 1000.0, 'EX_starch1200_m': 1000.0, 'EX_bglc_m': 1000.0}
```
after I built the community and modified the medium with
```
com = Community(data)
com.medium = rich_media
```
I got an error such as

```
WARNING  I could not find the following exchanges in community.py:668
                    your model: EX_cell4_m, EX_cell500_m,                      
                    EX_maltpt_m, EX_maltttr_m, EX_fe3_m,                        
                    EX_mn2_m, EX_mood_m, EX_cl_m, EX_cu2_m,                    
                    EX_malttr_m, EX_lnlnca_m, EX_ni2_m,                        
                    EX_starch2_m, EX_starch1200_m,                              
                    EX_amylose300_m, EX_cell6_m, EX_zn2_m,                      
                    EX_isomal_m, EX_lnlncg_m, EX_amylose_m,                    
                    EX_mg2_m, EX_bglc_m, EX_amylopect900_m,                    
                    EX_ca2_m, EX_starch1_m             
```
The excluded metabolites for each community are slightly different, but most of them (all the starches and complex sugars for example) are similar.

What is the meaning of this problem? Is the community cannot import those metabolites from the medium? Is there another way to stimulate growth on a rich medium?

Thank you

**Comment by cdiener** [comment, marked as answer]
Sorry for the delayed reply. An organism requires a transport reaction to be able to consume a particular metabolite. Sometimes an organism may not have the ability to do so. For instance, E. coli may not be able to break down or consume complex fibers even if they are present in the environment. In most cases this is to be expected and not a huge issue. The warning is there to be able to spot potential problems, for instance using different ID systems for the media and models (pretty common issue). It also means that everything not listed there *can be imported* by the community. If you still see realistic growth in the community I think it's not a large issue. I should definitely make this error message more informative or maybe only show it if a large amount of media components could not be imported.
